### PR TITLE
feat(redshift): column compression encodings and comments can now be customised

### LINKED ISSUE #22506: @aws-cdk/aws-redshift-alpha: Add encoding and comments for redshift tables column construct
### Describe the feature

Currently we have this construct:

https://docs.aws.amazon.com/cdk/api/v2/docs/@aws-cdk_aws-redshift-alpha.Table.html

Whose columns are defined as:

https://docs.aws.amazon.com/cdk/api/v2/docs/@aws-cdk_aws-redshift-alpha.Column.html

And this is great. However, in the world of big data, columns need encoding for faster performance and lower storage space:

https://docs.aws.amazon.com/redshift/latest/dg/c_Compression_encodings.html

Additionally, adding comments on columns allows us to be able document and version control our data definitions:

https://docs.aws.amazon.com/redshift/latest/dg/r_COMMENT.html

### Use Case

The current table construct allows the programmatic creation and modification of columns, however, in real life, econdings for performance and storage and comments for documentation are critical in any big data use case.

This addition will bring CDK one step closer to being able to fully manage a Redshift's table table structure and documentation.

### Proposed Solution

The idea, is that columns can be defined as:

```
{
  dataType: 'varchar(10)',
  name: 'my_col',
  distKey: false,
  sortKey: false,
  encoding: 'ZSTD',
  comment: 'some comment on column'
}
```

### Other Information

_No response_

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.45

### Environment details (OS name and version, etc.)

AmazonLinux2

### PR DESCRIPTION
**Pull request by Rizxcviii**
This feature request includes additions for compression encoding and comments for table columns. This feature request includes both features in one to close #22506

----

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md)

### Adding new Construct Runtime Dependencies:

* [ ] This PR adds new construct runtime dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md/#adding-construct-runtime-dependencies)

### New Features

* [x] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md)?
	* [x] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Review by TheRealAmazonKendra — DISMISSED** [review]
Thank you for your contribution! There's a lot going on in this PR and I'm inclined to think it needs to get separated into multiple. There's also a lot of duplication so we should see where we can reuse code instead adding it twice.

**Comment by Rizxcviii** [comment]
By duplication, are you referring to the duplication of the attributes for different interfaces/classes?
There seems to be this duplication throughout the redshift construct. In addition, the duplication of the `enum` is somewhat required as you cannot import it when the `enum` is located outside the private directory. Would there need to be a refactor of the entire redshift construct in that case?

**Comment by Rizxcviii** [comment]
> Thank you for your contribution! There's a lot going on in this PR and I'm inclined to think it needs to get separated into multiple. There's also a lot of duplication so we should see where we can reuse code instead adding it twice.

I've managed to solve this by importing the `enum` from the private folder, essentially exporting it afterwards, to allow for public use.

File: packages/@aws-cdk/aws-redshift/lib/private/database-query-provider/table.ts [hunk 1]
```diff
@@ -162,3 +181,10 @@ async function updateTable(
 function getSortKeyColumnsString(sortKeyColumns: Column[]) {
   return sortKeyColumns.map(column => column.name).join();
 }
+
+function getEncodingColumnString(column: Column): string {
+  if (column.encoding) {
+    return ` ENCODE ${column.encoding}`;
```
Comment (comcalvi) [inline packages/@aws-cdk/aws-redshift/lib/private/database-query-provider/table.ts:187]: minor, but could the space not be included? eg
```suggestion
    return `ENCODE ${column.encoding}`;
```

File: packages/@aws-cdk/aws-redshift/lib/private/database-query-provider/table.ts [hunk 2]
```diff
@@ -40,7 +40,7 @@ async function createTable(
   tableAndClusterProps: TableAndClusterProps,
 ): Promise<string> {
   const tableName = tableNamePrefix + tableNameSuffix;
-  const tableColumnsString = tableColumns.map(column => `${column.name} ${column.dataType}`).join();
+  const tableColumnsString = tableColumns.map(column => `${column.name} ${column.dataType}${getEncodingColumnString(column)}`).join();
```
Comment (comcalvi) [inline packages/@aws-cdk/aws-redshift/lib/private/database-query-provider/table.ts:43]:
```suggestion
  const tableColumnsString = tableColumns.map(column => `${column.name} ${column.dataType} ${getEncodingColumnString(column)}`).join();
```

File: packages/@aws-cdk/aws-redshift/test/integ.cluster-loggingbucket.js.snapshot/read.22836.lock [hunk 3]
```diff
@@ -0,0 +1 @@
+22836
```
Comment (comcalvi) [inline packages/@aws-cdk/aws-redshift/test/integ.cluster-loggingbucket.js.snapshot/read.22836.lock:1]: I'm pretty sure this shouldn't be in the snapshot

**Review by comcalvi — CHANGES_REQUESTED** [review]
This is really nice! Just a few comments

File: packages/@aws-cdk/aws-redshift/README.md [hunk 4]
```diff
@@ -200,17 +200,32 @@ new Table(this, 'Table', {
...
+```
+
+Table columns can be configured to use a specific compression encoding:
+
+```ts fixture=cluster
+import { ColumnEncoding } from '@aws-cdk/aws-redshift';
+
+new Table(this, 'Table', {
+  tableColumns: [
+    { name: 'col1', dataType: 'varchar(4)', encoding: ColumnEncoding.DELTA },
```
Comment (Rizxcviii) [inline packages/@aws-cdk/aws-redshift/README.md:224]: Note to self, probably change this to TEXT32K

File: packages/@aws-cdk/aws-redshift/lib/private/database-query-provider/table.ts (hunk 2 above)
Comment (Rizxcviii) [reply packages/@aws-cdk/aws-redshift/lib/private/database-query-provider/table.ts:43]: We could down this route, but it'll change the outputting text for a lot of strings, that append to this string. I've noticed that almost every test in the table test file needs to be changed to accommodate this change.

Comment (Rizxcviii) [reply packages/@aws-cdk/aws-redshift/lib/private/database-query-provider/table.ts]: This has been done

File: packages/@aws-cdk/aws-redshift/test/database-query-provider/table.test.ts [hunk 5]
```diff
@@ -92,7 +92,7 @@ describe('create', () => {
     await manageTable(newResourceProperties, event);
 
     expect(mockExecuteStatement).toHaveBeenCalledWith(expect.objectContaining({
-      Sql: `CREATE TABLE ${tableNamePrefix}${requestIdTruncated} (col1 varchar(1)) DISTSTYLE KEY DISTKEY(col1)`,
+      Sql: `CREATE TABLE ${tableNamePrefix}${requestIdTruncated} (col1 varchar(1) ) DISTSTYLE KEY DISTKEY(col1)`,
```
Comment (comcalvi) [inline packages/@aws-cdk/aws-redshift/test/database-query-provider/table.test.ts:95]: ah

I see.

I hate to do this, but can you revert that change to the spacing I asked in the first round? I hadn't expected it to result in extra spaces everywhere.

**Review by comcalvi — CHANGES_REQUESTED** [review]
Looks pretty much perfect, except for the odd spacing that occurs now...that's 100% my fault.

File: packages/@aws-cdk/aws-redshift/test/database-query-provider/table.test.ts (hunk 5 above)
Comment (Rizxcviii) [reply packages/@aws-cdk/aws-redshift/test/database-query-provider/table.test.ts]: lol, no problem. It was the last commit anyways, so I just reverted it.

**Review by comcalvi — APPROVED** [review]
Thanks for the PR! This is a neat change

**Pull request merged**